File: backend/src/Microservices/_Common/Common/Entity/Billing.cs
using MongoDB.Bson.Serialization.Attributes;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using TypeGen.Core.TypeAnnotations;

namespace Common.Entity
{
    [ExportTsInterface(OutputDir = "../../../../../frontend/web/parceldelivery-admin-app/src/app/_models")]
    public class Billing
    {
        public string Id { get; set; }
        public string UserId { get; set; }
        public string Name { get; set; }
        public double TotalDistance { get; set; }
        public double TotalAmount { get; set; }

        public int CurrencyId { get; set; }
        public Currency Currency { get; set; }

        [BsonIgnoreIfNull]
        public ShippingRequest ShippingRequest { get; set; }
    }
}

File: frontend/mobil/parceldelivery_mobile/lib/models/billing.dart
import 'package:freezed_annotation/freezed_annotation.dart';

import 'currency.dart';

part 'billing.freezed.dart';
part 'billing.g.dart';

@freezed
class Billing with _$Billing {
  const factory Billing({
    required String id,
    required String userId,
    required String name,
    required int totalDistance,
    required int totalAmount,
    required Currency currency,
  }) = _Billing;

  factory Billing.fromJson(Map<String, Object?> json) =>
      _$BillingFromJson(json);
}
